engine/table/src/main/java/io/deephaven/engine/table/impl/updateby/rollingformulamulticolumn/RollingFormulaMultiColumnOperator.java

```diff
@@ -0,0 +1,394 @@
+
+            // Create single value column sources of the appropriate type for each formula column input.
+            final Map<String, ColumnSource<?>> inputSources = new HashMap<>();
+            for (int i = 0; i < inputColumnNames.length; i++) {
+                final String inputColumnName = inputColumnNames[i];
+                final Class<?> inputColumnType = inputColumnTypes[i];
+                final Class<?> inputVectorType = inputVectorTypes[i];
+                final Class<?> inputComponentType = inputComponentTypes[i];
+
+                // Create and store the ring buffer for the input column.
+                final RingBuffer ringBuffer = RingBuffer.makeRingBuffer(
+                        inputColumnType,
+                        inputComponentType,
+                        BUFFER_INITIAL_CAPACITY,
+                        true);
+
+                // Create a single value column source wrapping the ring buffer.
+                // noinspection unchecked
+                final SingleValueColumnSource<Vector<?>> formulaInputSource =
+                        (SingleValueColumnSource<Vector<?>>) SingleValueColumnSource
+                                .getSingleValueColumnSource(inputVectorType);
+
+                final RingBufferVectorWrapper<?> wrapper = RingBufferVectorWrapper.makeRingBufferVectorWrapper(
+                        inputVectorType,
+                        inputComponentTypes[i],
+                        ringBuffer);
+
+                formulaInputSource.set(wrapper);
+
+                inputSources.put(inputColumnName, formulaInputSource);
+
+                inputConsumers[i] = RingBufferWindowConsumer.create(ringBuffer);
+            }
+            formulaCopy.initInputs(RowSetFactory.flat(1).toTracking(), inputSources);
+
+            formulaOutputSource = ReinterpretUtils.maybeConvertToPrimitive(formulaCopy.getDataView());
+            outputSetter = getChunkSetter(outputValues, formulaOutputSource);
+        }
+
+        @Override
+        protected void setValueChunks(@NotNull Chunk<? extends Values>[] valueChunks) {
+            // Assign the influencer values chunks to the input consumers.
+            for (int i = 0; i < valueChunks.length; i++) {
+                inputConsumers[i].setInfluencerValuesChunk(valueChunks[i]);
+            }
+        }
+
+        @Override
+        protected void push(int pos, int count) {
+            throw new IllegalStateException("RollingFormulaMultiColumnOperator.Context.push should never be called.");
+        }
+
+        @Override
+        public void accumulateCumulative(
+                @NotNull final RowSequence inputKeys,
+                @NotNull final Chunk<? extends Values>[] valueChunkArr,
+                @Nullable final LongChunk<? extends Values> tsChunk,
+                final int len) {
+            throw new UnsupportedOperationException("RollingFormula is not supported in cumulative operations.");
+        }
```

Comment:
Is it all RollingFormula or just this one?
```suggestion
            throw new UnsupportedOperationException("RollingFormulaMultiColumnOperator is not supported in cumulative operations.");
        }
```

Reply:
No `Rolling` operator is defined for cumulative operations.